src/components/Image/Image.js

```diff
@@ -36,19 +35,56 @@ function collect(connect, monitor) {
 class Image extends Component {
   constructor(props) {
     super(props);
-
     this.state = {
-      data: ''
+      externalData: null
```

Comment:
I would use `src`. It will make more sense when using the element, e.g. `<img src={this.src}>`.